packages/dcos-integration-test/extra/conftest.py

```diff
@@ -42,3 +49,45 @@ def dcos_api_session():
 @pytest.fixture(scope='session')
 def noauth_api_session(dcos_api_session):
     return dcos_api_session.get_user_session(None)
+
+
+@pytest.fixture(scope='session', autouse=True)
+def _dump_diagnostics(request, dcos_api_session):
+    """Download the zipped diagnostics bundle report from each master in the cluster to the home directory. This should
+    be run last. The _ prefix makes sure that pytest calls this first out of the autouse session scope fixtures, which
+    means that its post-yield code will be executed last.
+
+    * There is no official way to ensure fixtures are called in a certain order
+    https://github.com/pytest-dev/pytest/issues/1216
+    * However it seems that fixtures at the same scope are called alphabetically
+    https://stackoverflow.com/a/28593102/1436300
+    """
+    yield
+
+    if os.environ.get('DIAGNOSTICS_DIRECTORY'):
```

Comment:
Should documentation be updated with this env var? 

What are the values expected? Should the value returned be verified?

Reply:
`DIAGNOSTICS_DIRECTORY` is an environment variable used in TeamCity or whatever is calling into pytest to indicate where the job should scp the diagnostics report to from the cluster. In the fixture it's just used as an indicator of whether or not the diagnostics job should be run and the value doesn't get fed into anythign else. I didn't want to set an extra environment variable when the directory being set was an indicator of whether to run it, but I can change that to
```
make_diagnostics_report = os.environ.get('DIAGNOSTICS_DIRECTORY') is not None
if make_diagnostics_report:
```
unless you think it's better to just make another env var for `RUN_DIAGNOSTICS`? I think the latter is cleaner from the POV of dcos-integration-tests but from the CI setup it's another variable that we have to remember to update when changing things.

Reply:
Thanks for the response.

I agree with keeping it a single env var and not breaking it up.